File: tests/bindings/java/foo-tests/src/test/java/io/stepfunc/foo_test/DurationTest.java
package io.stepfunc.foo_test;

import io.stepfunc.foo.DurationEchoFunctions;
import org.junit.jupiter.api.Test;

import java.time.Duration;

import static org.assertj.core.api.Assertions.assertThat;

public class DurationTest {
    @Test
    public void DurationZeroTest() {
        Duration zero = Duration.ZERO;
        assertThat(DurationEchoFunctions.millisecondsEcho(zero)).isEqualTo(zero);
        assertThat(DurationEchoFunctions.secondsEcho(zero)).isEqualTo(zero);
        assertThat(DurationEchoFunctions.secondsFloatEcho(zero)).isEqualTo(zero);
    }

    @Test
    public void Duration5sTest() {
        Duration test = Duration.ofSeconds(5);
        assertThat(DurationEchoFunctions.millisecondsEcho(test)).isEqualTo(test);
        assertThat(DurationEchoFunctions.secondsEcho(test)).isEqualTo(test);
        assertThat(DurationEchoFunctions.secondsFloatEcho(test)).isEqualTo(test);
    }

    @Test
    public void Duration250msTest() {
        Duration test = Duration.ofMillis(250);
        assertThat(DurationEchoFunctions.millisecondsEcho(test)).isEqualTo(test);
        assertThat(DurationEchoFunctions.secondsEcho(test)).isEqualTo(Duration.ZERO);
        assertThat(DurationEchoFunctions.secondsFloatEcho(test)).isEqualTo(test);
    }

    @Test
    public void Duration41DaysTest() {
        Duration test = Duration.ofDays(41);
        assertThat(DurationEchoFunctions.millisecondsEcho(test)).isEqualTo(test);
        assertThat(DurationEchoFunctions.secondsEcho(test)).isEqualTo(test);
        assertThat(DurationEchoFunctions.secondsFloatEcho(test)).isEqualTo(test);
    }
}

File: tests/bindings/dotnet/foo.Tests/DurationTest.cs
using System;
using Xunit;
using foo;

namespace foo.Tests
{
    public class DurationTest
    {
        [Fact]
        public void DurationZeroTest()
        {
            var zero = TimeSpan.Zero;
            Assert.Equal(zero, DurationEchoFunctions.MillisecondsEcho(zero));
            Assert.Equal(zero, DurationEchoFunctions.SecondsEcho(zero));
            Assert.Equal(zero, DurationEchoFunctions.SecondsFloatEcho(zero));
        }

        [Fact]
        public void Duration5sTest()
        {
            var fiveS = TimeSpan.FromSeconds(5);
            Assert.Equal(fiveS, DurationEchoFunctions.MillisecondsEcho(fiveS));
            Assert.Equal(fiveS, DurationEchoFunctions.SecondsEcho(fiveS));
            Assert.Equal(fiveS, DurationEchoFunctions.SecondsFloatEcho(fiveS));
        }

        [Fact]
        public void Duration250msTest()
        {
            var test = TimeSpan.FromMilliseconds(250);
            Assert.Equal(test, DurationEchoFunctions.MillisecondsEcho(test));
            Assert.Equal(TimeSpan.FromSeconds(0), DurationEchoFunctions.SecondsEcho(test));
            Assert.Equal(test, DurationEchoFunctions.SecondsFloatEcho(test));
        }

        [Fact]
        public void Duration41DaysTest()
        {
            var test = TimeSpan.FromDays(41);
            Assert.Equal(test, DurationEchoFunctions.MillisecondsEcho(test));
            Assert.Equal(test, DurationEchoFunctions.SecondsEcho(test));
            Assert.Equal(test, DurationEchoFunctions.SecondsFloatEcho(test));
        }
    }
}
